Sanitation setup

rakaar opened the pull request:
Pylama, isort and mypy are setup. The same tools are used in pre-commit hook which can be checked at `.pre-commit-config.yaml` file. These tools are set up with travis CI which can be checked  at `.travis.yml` file.

concerns #7

thealphadollar commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,13 @@
+repos:
```
This is something new and looks dope 😉 I'll take some time to review this.

thealphadollar commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+language: python
+python: -"3.6"
```
Let's use 3.7

rakaar replied:
should there be any change is `Pipfile` , where  it is [requires]python_version="3.6"

thealphadollar replied:
Thanks.

thealphadollar commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+script:
+  - pylama
+  - isort -rc --check-only
+  - mypy --check-untyped-defs
```
create a `mypy.ini` config file?

thealphadollar commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Please remove all instances of `.vscode`

thealphadollar replied:
Thanks :D

thealphadollar commented on Pipfile:
```diff
@@ -21,6 +21,10 @@ pytz = "==2019.1"
 six = "==1.12.0"
 sqlalchemy = "==1.3.5"
 werkzeug = "==0.15.4"
+pylama = "*"
```
Please create config files for the sanity checkers as well.

thealphadollar replied:
Thanks.

thealphadollar commented:
Please setup logging as well :D

thealphadollar commented:
Yes :)

On Tue, Jul 2, 2019 at 1:48 PM Raghavendra Kaushik <notifications@github.com>
wrote:

> *@rakaar* commented on this pull request.
> ------------------------------
>
> In .travis.yml
> <https://github.com/iit-technology-ambit/common-backend/pull/11#discussion_r299357344>
> :
>
> > @@ -0,0 +1,12 @@
> +language: python
> +python: -"3.6"
>
> should there be any change is Pipfile , where it is
> [requires]python_version="3.6"
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/iit-technology-ambit/common-backend/pull/11?email_source=notifications&email_token=AH2K2IGWAGP6JSPUQNPVZ7TP5MFMZA5CNFSM4H4Y7Z4KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB5GW5IA#discussion_r299357344>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH2K2IECBENOZIQ2FTASNN3P5MFMZANCNFSM4H4Y7Z4A>
> .
>

rakaar commented:
1)Removed the .vscode folder
2)changed Python version to 3.7 in `Pipfile` and `.travis.yml` file
3)Conifguration files for sanity tests:
- pylama : `pylama.ini` (as in README.md of pylama project)
- isort: `.isort.cfg` (as in README.md of isort project)
- mypy: `mypy.ini` (as in Sample Platform of CC Extractor)

thealphadollar commented on .travis.yml:
```diff
@@ -1,5 +1,7 @@
 language: python
-python: -"3.6"
+python: -"3.7"
+  -"nightly"
```
We don't need to test on nightly.

thealphadollar commented on .isort.cfg:
```diff
@@ -0,0 +1,14 @@
+[settings]
+line_length=120
+force_to_top=file1.py,file2.py
```
We do not require anything below line 2 (don't remove line 2) at the present.

thealphadollar commented on pylama.ini:
```diff
@@ -0,0 +1,5 @@
+format = pylint
+skip = */.tox/*,*/.env/*
+linters = pylint,mccabe
+ignore = F0401,C0111,E731
```
Remove line number 5

thealphadollar commented:
@rakaar Can you please add information about this in a new file CONTRIBUTING.md as well? 

For an example on how to write about them, see Etiquettes under Contributing [here in sample-platform README.md](https://github.com/CCExtractor/sample-platform/blob/master/README.md)

Also add the instructions that one should run `pre-commit install` as the first thing after installing dependencies.

thealphadollar commented on .travis.yml:
```diff
@@ -0,0 +1,13 @@
+script:
+  - pylama
+  - isort -rc --check-only
+  - mypy --check-untyped-defs
```
Please set --check-untyped-defs in mypy config itself and do not put anything here.

thealphadollar replied:
@rakaar Implement this please.

rakaar commented:
@thealphadollar , Can you tell more about what should be present in `Contributing.md` for this proj

thealphadollar commented on .mypy_cache/3.7/_ast.meta.json:
```diff
@@ -0,0 +1 @@
+{"child_modules": [], "data_mtime": 1562160210, "dep_lines": [1, 2, 1, 1], "dep_prios": [10, 5, 5, 30], "dependencies": ["sys", "typing", "builtins", "abc"], "hash": "e4cd8a8eaf09e16e632732ea78340caa", "id": "_ast", "ignore_all": true, "interface_hash": "2099247ff3af08c270f478877c1f0148", "mtime": 1562042558, "options": {"allow_redefinition": false, "allow_untyped_globals": false, "always_false": ...
```
Please add `.mypy_cache` to .gitignore

thealphadollar commented:
@rakaar I've attached a template contributing.md you can use. Our main purpose should be to convey what all standard we follow in the code as well as the organisation. You should mention that our code passes through x, y and z checks (in bullet form as in template) and what they ensure. Also, we should mention how we ensure that these are enforced (using pre-commit hook and CI tests).

Use your own creativity and I'll improve it when you send in the PR :smile: 

https://gist.github.com/PurpleBooth/b24679402957c63ec426

thealphadollar commented:
Thank you. Looks great to me :D
